modules/core/src/main/java/org/apache/ignite/internal/processors/metastorage/persistence/DistributedMetaStorageHistoryItem.java

```diff
@@ -68,6 +78,31 @@ public long estimateSize() {
         return size;
     }
 
+    /** {@inheritDoc} */
+    @Override protected void writeExternalData(ObjectOutput out) throws IOException {
+        out.writeInt(keys.length);
+
+        for (int i = 0; i < keys.length; i++) {
+            U.writeString(out, keys[i]);
+
+            U.writeByteArray(out, valBytesArray[i]);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    protected void readExternalData(byte protoVer, ObjectInput in) throws IOException {
+        int len = in.readInt();
+
+        keys = new String[len];
```

Comment:
Do we really need empty arrays in case if `len == 0`?

Reply:
len == 0 is impossible, I'll assert it in the constructor.